### PR TITLE
UN-779: Article tags on Focused Article and Record Articles

### PR DESCRIPTION
**Pull request by jamesbiggs**
Ticket URL: [UN-779]

## About these changes

Added "article tags" to Focused and Record Article pages.
This allows pages to be tagged with the same tags as the Article pages.

Moved the tagging field to "promote" panels as per the ticket.

Search fields are working so editors can search by tags and get Focused, Record, and normal Articles in the search bar.

I tried moving the `ArticleTagMixin` to the `core.mixins.py` but this would've required extra migrations - but I think it's best to keep it in the `articles` app as I can't see it being used outside of this.

## How to check these changes

Open any of the pages under "Stories from the collection" and ensure that Article tags can be added.

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer.
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [x] Added/updated tests and documentation where relevant.

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

[UN-779]: https://national-archives.atlassian.net/browse/UN-779?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: etna/articles/models.py [hunk 1]
```diff
@@ -41,6 +41,73 @@
...
+
+class TaggedArticle(ItemBase):
+    tag = models.ForeignKey(
+        ArticleTag, related_name="tagged_article", on_delete=models.CASCADE
+    )
+    content_object = ParentalKey(
+        to="wagtailcore.Page",
+        on_delete=models.CASCADE,
+        related_name="tagged_items",
+    )
```
Comment (TNA-Allan) [inline etna/articles/models.py:90]: Was it necessary to move these methods above? With changes like these it appears to be new ones to review. (only  to="wagtailcore.Page", has changed though)

I think, previously these methods are under ArticleIndexPage as it does which do not use them, and seemed logically placed. 

Will leave it to you if you if you'd like to restore them to their original location.

File: etna/articles/migrations/0095_focusedarticlepage_article_tag_names_and_more.py
Comment (TNA-Allan) [inline etna/articles/migrations/0095_focusedarticlepage_article_tag_names_and_more.py:1]: By the time, I've reviewed this, the latest base changes have  095_alter_articlepage_body_alter_focusedarticlepage_body.py. Do you want to sync with latest for a new sequence?

**Review by TNA-Allan — CHANGES_REQUESTED** [review]

File: etna/articles/migrations/0095_focusedarticlepage_article_tag_names_and_more.py
Comment (jamesbiggs) [reply etna/articles/migrations/0095_focusedarticlepage_article_tag_names_and_more.py:1]: Hi Allan, yes I shall do this :)

File: etna/articles/models.py (hunk 1 above)
Comment (jamesbiggs) [reply etna/articles/models.py:90]: Hi Allan, I think to me it just made sense to have everything at the top but I do get your point 🙂 I think I will leave it there but if you'd like me to move it, then I can - happy for whatever you think🙂

Comment (TNA-Allan) [reply etna/articles/models.py:90]: I have conveyed my view and As  I mentioned earlier left it to you if you wanted to restore it.

**Pull request merged**